engine/table/src/main/java/io/deephaven/engine/util/GroovyDeephavenSession.java

```diff
@@ -270,9 +366,9 @@ private Exception maybeRewriteStackTrace(String scriptName, String currentScript
         return e;
     }
 
-    private static Class<?> loadClass(String className) throws ClassNotFoundException {
+    private Class<?> loadClass(String className) throws ClassNotFoundException {
         try {
-            return Class.forName(className, false, GroovyDeephavenSession.class.getClassLoader());
+            return Class.forName(className, false, this.groovyShell.getClassLoader());
```

Comment:
As above, a custom classloader installed by the app before calling this wouldn't necessarily be called when a class needs to be loaded in this way. Between this and the above change, not only does groovy get a chance to compile if needed, but a custom classloader can also be used.